### ISSUE TITLE
Stack overflow error when opening chest.

### ISSUE THREAD

**Issue by lucifer6642** [SimpleServer/SimpleServer#258]
World was created in 1.2.3 and running fine on simpleserver compiled from git on 05-Apr-2012.

Since issue 250 was resolved, cloned, edited version so it would work with 1.2.5 and recompiled yesterday afternoon.

Tried to open a chest and got the following 

```
Exception in thread "ServerStream-Lucifer_LLS" java.lang.StackOverflowError
        at simpleserver.Coordinate.add(Unknown Source)
        at simpleserver.Coordinate.add(Unknown Source)
        at simpleserver.Coordinate.add(Unknown Source)
        at simpleserver.Coordinate.add(Unknown Source)
        at simpleserver.Coordinate.add(Unknown Source)
        at simpleserver.Coordinate.add(Unknown Source)
```

Although I get 1024 of those Coordinate lines.  Shortly afterwards, client will be disconnected

```
java.net.SocketTimeoutException: Read timed out
[SimpleServer] java.io.EOFException
[SimpleServer] ServerStream error handling traffic for <masked IP> (23)
2012-04-26 04:55:37 [INFO] Lucifer_LLS lost connection: disconnect.genericReason

[SimpleServer] Socket Closed: <masked IP>
```

---edit---

Replacing with older simpleserver file results in many "Skipping corrupt chest" messages. Had to restore older simpleserver.dat file

**Comment by apirogov** [comment]
Hmmm it might be a bug in the conversion of the chest coordinate saving from byte to int .. :/ Which is in the fix for 251... hmm tricky one, I'm not an expert on NBT data...

**Comment by apirogov** [comment]
Should be fixed in 4d655da
